etcdserver/v3_server.go

```diff
@@ -639,7 +639,7 @@ func (s *EtcdServer) isValidSimpleToken(token string) bool {
 	}
 }
 
-func (s *EtcdServer) authInfoFromCtx(ctx context.Context) (*auth.AuthInfo, error) {
+func (s *EtcdServer) AuthInfoFromCtx(ctx context.Context) (*auth.AuthInfo, error) {
```

Comment:
it seems like this belongs in the auth package if it's important enough to export? can probably pass a `func f(idx uint64) <-chan` to get around the etcdserver dependency in isValidToken where it waits for the revision update